pabi_procurement/models/purchase_request.py

```diff
@@ -85,26 +96,59 @@ class PurchaseRequest(models.Model):
     @api.model
     def create(self, vals):
         create_rec = super(PurchaseRequest, self).create(vals)
-        sum_total = 0.0
+        AccTax = self.env['account.tax']
+        total_untaxed = 0.0
+        tax_amount = 0.0
         if 'line_ids' in vals:
             if len(vals['line_ids']) > 0:
                 for line_rec in vals['line_ids']:
                     field = line_rec[2]
-                    sum_total += field['product_qty'] * field['price_unit']
-            create_rec.amount_total = sum_total
+                    for rec_taxes in field['taxes_id']:
+                        for rec_tax in rec_taxes[2]:
+                            domain = [('id', '=', rec_tax)]
+                            found_tax = AccTax.search(domain)
+                            if found_tax.type == 'percent':
+                                tax_amount += field['product_qty'] * (
+                                    field['price_unit'] * found_tax.amount
+                                )
+                            elif found_tax.type == 'fixed':
+                                tax_amount += field['product_qty'] * (
+                                    field['price_unit'] + found_tax.amount
+                                )
+                    untaxed = field['product_qty'] * field['price_unit']
+                    total_untaxed += untaxed
+            create_rec.amount_untaxed = total_untaxed
```

Comment:
Beware about Performance. This is 3 write statement. Instead use write() once. is better.
+            create_rec.amount_untaxed = total_untaxed
 +            create_rec.amount_tax = tax_amount
 +            create_rec.amount_total = total_untaxed + tax_amount

But actually, why do we need to write at all? Why don't just calculate all value and put it to create(vals) ???